src/lake/Lake/Build/Monad.lean

```diff
@@ -47,10 +47,46 @@ If a cycle is encountered, log it and then fail.
 @[inline] def RecBuildM.run (build : RecBuildM α) : BuildM α := do
   (·.1) <$> build.runIn {}
 
+/--
+Busy waits to acquire the lock represented by the `lockFile`.
+Prints a warning if on the first time it has to wait.
+-/
+@[inline] partial def busyAcquireLockFile (lockFile : FilePath) : IO PUnit := do
+  busyLoop true
+where
+  busyLoop firstTime :=
+    try
+      -- Remark: fail if already exists
+      -- (not part of POSIX, but supported on all our platforms)
+      createParentDirs lockFile
+      let h ← IO.FS.Handle.mk lockFile .writeNew
+      h.putStrLn <| toString <| ← IO.Process.getPID
```

Comment:
So the PID doesn't do anything yet?

Reply:
Not yet, Lake does not use it. However, it useful for users to know what process is holding the lock (or was, if `lake` died ungracefully).